### PR TITLE
MLT-0039 Deleting orders

### PR DESCRIPTION
**Pull request by MormonJesus69420**
Allows for order deletion, not fully tested or cleaned up as we will have a workshop tomorrow for this project and will likely rework some functionality.

### CODE REVIEW

File: src/main/kotlin/no/nb/mlt/wls/order/payloads/SynqOrderPayload.kt [hunk 1]
```diff
@@ -32,13 +31,8 @@ data class SynqOrderPayload(
...
 
-    enum class SynqOrderType(private val type: String) {
-        STANDARD("Standard");
-
-        @JsonValue
-        override fun toString(): String {
-            return type
-        }
+    enum class SynqOrderType {
+        STANDARD
```
Comment (anotheroneofthese) [inline src/main/kotlin/no/nb/mlt/wls/order/payloads/SynqOrderPayload.kt:35]: This is likely a merge error, as this code is required for SynQ to properly accept our payloads

File: src/main/kotlin/no/nb/mlt/wls/order/controller/OrderController.kt [hunk 2]
```diff
@@ -164,4 +164,39 @@ class OrderController(val orderService: OrderService) {
...
+    )
+    @ApiResponses(
+        ApiResponse(
+            responseCode = "200",
+            description = "Order with given 'hostName' and 'hostOrderId' was deleted from the system.",
+            content = [Content(schema = Schema())]
+        ),
+        ApiResponse(
+            responseCode = "401",
+            description = "Client sending the request is not authorized order products."
```
Comment (anotheroneofthese) [inline src/main/kotlin/no/nb/mlt/wls/order/controller/OrderController.kt:182]:
```suggestion
            description = "Client sending the request is not authorized to order products."
```

**Review by anotheroneofthese — CHANGES_REQUESTED** [review]

File: src/main/kotlin/no/nb/mlt/wls/order/payloads/SynqOrderPayload.kt (hunk 1 above)
Comment (MormonJesus69420) [reply src/main/kotlin/no/nb/mlt/wls/order/payloads/SynqOrderPayload.kt:35]: Oh yeah might have missed that, thanks for finding it

File: src/main/kotlin/no/nb/mlt/wls/order/service/SynqOrderService.kt [hunk 3]
```diff
@@ -57,20 +57,38 @@ class SynqOrderService(
     }
 
     suspend fun updateOrder(payload: ApiUpdateOrderPayload): ResponseEntity<SynqError> {
-        val uri = URI.create("$baseUrl/orders/batch")
-
+        // TODO - Should maybe just get a regular SynqOrderPayload here instead of converting it?
+        // If anything the regular order service should do the conversion either at call site or somewhere else
         val orders = SynqOrder(listOf(payload.toOrder().toSynqPayload()))
```
Comment (anotheroneofthese) [inline src/main/kotlin/no/nb/mlt/wls/order/service/SynqOrderService.kt:62]: I agree, it makes sense to move this out of the SynQ handling

Comment (MormonJesus69420) [reply src/main/kotlin/no/nb/mlt/wls/order/service/SynqOrderService.kt:62]: True but we can look at it during workshop today

**Review by anotheroneofthese — APPROVED** [review]

**Pull request merged**